thompson_sampling/thompson_strategy.py

```diff
@@ -78,3 +89,51 @@ def mean(self):
     @property
     def selections(self):
         return self.num_selections
+
+
+class GaussianProcessesStrategy(ThompsonStrategy):
+    def __init__(
+        self,
+        num_parameters: int,
+        interval: "np.ndarray | list[np.ndarray]",
+        beta: float,
+    ):
+        super().__init__()
+        self.num_parameters = num_parameters
+        if isinstance(interval, list):
+            if len(interval) != self.num_parameters:
+                raise ValueError(
+                    "When passing a list the length should be the same as num_parameters"
+                )
+            else:
+                self.mesh = np.meshgrid(*interval)
+
+        elif isinstance(interval, np.ndarray):
+            self.mesh = np.meshgrid(*[interval for _ in range(self.num_parameters)])
+
+        mapfunc = partial(np.ravel, order="A")
+        self.X_grid = np.column_stack(list(map(mapfunc, self.mesh)))
+
+        self.mu = np.array([0.0 for _ in range(self.X_grid.shape[0])])
+        self.sigma = np.array([np.random.randn() for _ in range(self.X_grid.shape[0])])
+        self.beta = beta
+        kernel = C(1, (1e-3, 1e1)) * RBF(1, (1e-2, 1e1)) + WhiteKernel()
```

Comment:
I believe a better option is to pass the kernel to this class. in this way you don't have to define it here. Maybe, even better, a user can define its own gaussian process regressor and pass it to this function.

Reply:
fixed in 94ba55000419022918e29b985e505e5bc109131b